### ISSUE TITLE
Sorting of fields in field visualization component

### ISSUE THREAD

**Issue by lklic** [researchspace/researchspace#19]
Opening a feature request to be able to sort the fields in the field visualization component.

Currently there is no way to sort fields in this component to be able to control the order in which they show up when listing fields based on the domain and range. We have done it manually by writing sparql querries agains the field definition repository, but it would be better to have a way to control this within a field definition.

This proposal is to add a numerical field "weight" to the field definition.

This way you can update the E22 template as such to see the list ordered:
```
            <mp-field-visualization
              subject=[[this]]
              fields='[[fieldDefinitionsFromQuery "SELECT ?field WHERE { ?field <http://www.metaphacts.com/ontology/fields#domain> crm:E22_Man-Made_Object. OPTIONAL {?field <https://schema.org/weight> ?weight}} ORDER BY ASC(?weight)"]]'
              template='{{> rsp:FieldBasedVisualization subject="[[this]]" [[#if asserted]]guess="true"[[/if]]}}'
            ></mp-field-visualization>
```

The <https://schema.org/weight> property seems to be a good predicate to describe this.

We will create a pull request tomorrow with a proposed implementation of this.

**Comment by doldman** [comment]
Yes, this has been something that keeps on falling off the edge but is something that eveyone needs. One issue here is that different templates may make use of overallaping patterns and require different orders. For class based templates it would be good to have an order for a class to reflect the different relationship of patterns. Is this something that might be added in the future?

**Comment by lklic** [comment]
We are going to make a pull request for this tomorrow as we have already implemented it. Currently we are doing all of the testing...
I think for the use case of having different sort order for different templates we can combine them with field categories, which is another pull request we will also make, together with a category manager to manage them

**Comment by metaschell** [comment]
We have merged PR #34 locally so this issue can be closed. Thanks for the contribution!
It will become available with the next update of this repo (as part of the 3.3 release).

Note: having a `order` field globally for all fields helps to get some initial support in ordering fields in a form.
But this is only a partial solution as one would typically rather want views which define a set of fields to be applied in a certain order, e.g. for a specific class. Also, there might be different views for different use case for a class (or not even per class, but in a more general case). So to evolve this further we need to discuss this a bit more. Your input and use cases are very welcome here.
